iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffMatchingRetainedAffPosQualityTest.java

```diff
@@ -0,0 +1,359 @@
+        
+        affMatchingService.matchAffiliations(sparkContext, inputAffDirPath, inputOrgDirPath, outputDirPath);
+        
+        
+        // print results quality
+        
+        System.out.println("\nDOCUMENTS ASSIGNED TO PROJECT");
+        
+        readResultsAndPrintQualityRate(of("src/test/resources/experimentalData/expectedOutput/set2/matched_aff.json"));
+    }
+    
+    
+    @Test
+    public void matchAffiliations_combined_data() throws IOException {
+        
+        // given
+        
+        createInputDataFromJsonFiles(
+                of("src/test/resources/experimentalData/input/all_organizations.json"),
+                of(
+                        "src/test/resources/experimentalData/input/set1/docs_with_aff_real_data.json",
+                        "src/test/resources/experimentalData/input/set2/docs_with_aff_real_data.json"),
+                of(), of());
+        
+        
+        // execute
+        
+        affMatchingService.matchAffiliations(sparkContext, inputAffDirPath, inputOrgDirPath, outputDirPath);
+        
+        
+        // print results quality
+        
+        System.out.println("\nALL TEST DATA");
+        
+        readResultsAndPrintQualityRate(
+                of(
+                        "src/test/resources/experimentalData/expectedOutput/set1/matched_aff.json",
+                        "src/test/resources/experimentalData/expectedOutput/set2/matched_aff.json"));
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private void createInputDataFromJsonFiles(List<String> jsonInputOrgPaths, List<String> jsonInputAffPaths, List<String> jsonInputDocProjPaths, List<String> jsonInputProjOrgPaths) throws IOException {
+        
+        createLocalAvroDataStore(readMultipleJsonDataStores(jsonInputOrgPaths, Organization.class), inputOrgDirPath, Organization.class);
+        createLocalAvroDataStore(readMultipleJsonDataStores(jsonInputAffPaths, ExtractedDocumentMetadata.class), inputAffDirPath, ExtractedDocumentMetadata.class);
+        
+        createLocalAvroDataStore(readMultipleJsonDataStores(jsonInputDocProjPaths, DocumentToProject.class), inputDocProjDirPath, DocumentToProject.class);
+        createLocalAvroDataStore(readMultipleJsonDataStores(jsonInputProjOrgPaths, ProjectToOrganization.class), inputProjOrgDirPath, ProjectToOrganization.class);
+        
+    }
+    
+    private void readResultsAndPrintQualityRate(List<String> expectedResultsJsonPaths) throws IOException {
+        
+        List<SimpleAffMatchResult> actualMatches = readJson(outputDirPath + "/part-00000", SimpleAffMatchResult.class);
+        List<SimpleAffMatchResult> expectedMatches = readMultipleJsons(expectedResultsJsonPaths, SimpleAffMatchResult.class);
+        
+        
+        printTruePositivesFactor(expectedMatches, actualMatches);
+        printFalsePositivesFactor(expectedMatches, actualMatches);
+        if (PRINT_FALSE_POSITIVE_MATCHES) {
```

Comment:
Do we need to check this condition (and the one below) since `boolean` field is final and will never be changed? I guess this field was created to simplify executions triggered in 'development' phase to easily change generated output, am I right?